nsv13/code/modules/collision/vector2d.dm

```diff
@@ -0,0 +1,151 @@
+/*
+
+Vector class written by Monster860, modified by Kmc2000 under Qwerty's guidance.
+
+This uses operator overloading
+
+Special thanks to qwertyquerty for explaining and dictating a lot of this! (I've mostly translated his pseudocode into readable byond code)
+
+*/
+
+/datum/vector2d
+	var/x
+	var/y
+
+/*
+Constructor for vector2d objects, taking a simple X,Y coordinate.
+*/
+
+/datum/vector2d/New(x = 0, y = 0)
+	src.x = x
+	src.y = y
+	..()
+
+/*
+Method to turn this vector counter clockwise by a desired angle
+*/
+
+/datum/vector2d/proc/rotate(angle)
+	var/s = sin(angle)
+	var/c = cos(angle)
+	newx = c*x + s*y
+	newy = -s*x + c*y
```

Comment:
wait wtf how does this not error.....

Reply:
It does. This is a draft of a PR. She just put it up so I can review as she writes